src/ApiManagement/ApiManagement.ServiceManagement/Commands/UpdateAzureApiManagementGateway.cs

```diff
@@ -0,0 +1,75 @@
+﻿//  
+// Copyright (c) Microsoft.  All rights reserved.
+// 
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//    http://www.apache.org/licenses/LICENSE-2.0
+// 
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+namespace Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Commands
+{
+    using Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Models;
+    using Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Properties;
+    using System;
+    using System.Management.Automation;
+
+    [Cmdlet("Update", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ApiManagementGateway", SupportsShouldProcess = true)]
+    [OutputType(typeof(PsApiManagementGateway))]
+    public class UpdateAzureApiManagementGateway : AzureApiManagementCmdletBase
```

Comment:
Please support pipe for this cmdlet.

Reply:
@vfedonkin to support pipe for this cmdlet, we need to add support for `-InputObject`  or `-ResourceId` input parameter. Please refer to details [here](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/design-guidelines/piping-best-practices.md#using-the--inputobject-parameter)

Refer to example [here](https://github.com/Azure/azure-powershell/blob/master/src/ApiManagement/ApiManagement.ServiceManagement/Commands/UpdateAzureApiManagementCache.cs#L55)

Reply:
Done